presentation/src/main/java/com/exallium/mvvmcleanexample/presentation/users/UserEditViewModel.kt

```diff
@@ -1,61 +1,92 @@
 package com.exallium.mvvmcleanexample.presentation.users
 
-import android.databinding.ObservableField
+import android.databinding.Bindable
 import com.exallium.mvvmcleanexample.domain.UseCaseResult
 import com.exallium.mvvmcleanexample.domain.users.SaveUserUseCase
 import com.exallium.mvvmcleanexample.domain.users.User
 import com.exallium.mvvmcleanexample.domain.users.ValidateUserFirstNameUseCase
 import com.exallium.mvvmcleanexample.domain.users.ValidateUserLastNameUseCase
 import com.exallium.mvvmcleanexample.presentation.nav.UiRouter
+import com.exallium.mvvmcleanexample.presentation.utils.bindable
+import com.exallium.mvvmcleanexample.presentation.utils.delegates.DisposableProperty
+import com.exallium.mvvmcleanexample.presentation.utils.distinctObservable
+import com.exallium.mvvmcleanexample.presentation.utils.disposable
+import com.exallium.mvvmcleanexample.presentation.utils.notifiable.BaseNotifiableObservable
+import com.exallium.mvvmcleanexample.presentation.utils.notifiable.NotifiableObservable
+import com.exallium.mvvmcleanexample.presentation.utils.onNext
 import com.jakewharton.rxrelay2.BehaviorRelay
 import com.jakewharton.rxrelay2.PublishRelay
-import io.reactivex.Observable
-import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
 
-class UserEditViewModel(userEditModel: UserEditModel,
-                        private val router: UiRouter) {
+class UserEditViewModel(
+        private val userEditModel: UserEditModel,
+        private val router: UiRouter,
+        private val firstNameRelay: BehaviorRelay<String> = BehaviorRelay.create(),
+        private val lastNameRelay: BehaviorRelay<String> = BehaviorRelay.create(),
+        private val savedUserRelay: PublishRelay<User> = PublishRelay.create(),
+        disposableProperty: DisposableProperty = disposable(null),
+        notifiableObservable: NotifiableObservable = BaseNotifiableObservable())
+    : NotifiableObservable by notifiableObservable {
 
-    val firstNameError = ObservableField<String>("")
-    val lastNameError = ObservableField<String>("")
+    init {
+        setDelegator(this)
+    }
 
-    private val compositeDisposable = CompositeDisposable()
+    /* symbol meaning:
+    >> : bind direction left to right, future changes are propagated
+    << : bind direction right to left, future changes are propagated
+    >| : bind direction left to right, future changes are not propagated
+    |< : bind direction right to left, future changes are not propagated
+     */
 
-    private val clickRelay = PublishRelay.create<User>()
-    private val firstNameRelay = BehaviorRelay.create<String>()
-    private val lastNameRelay = BehaviorRelay.create<String>()
+    // view |< >> view model
+    var firstName: String by distinctObservable(
+            initialValue = "",
```

Comment:
The view will take the initial value from the `firstName` property, but if the view model changes this property it will not be propagated to the view. Once the initial value has been read by the view (view binding passes it to the view I mean), the view (binding) will simply write to the property, and the view model will observe the changes.